apollo-router-core/src/naive_introspection.rs

```diff
@@ -47,26 +47,42 @@ impl NaiveIntrospection {
             schema.as_str(),
             KNOWN_INTROSPECTION_QUERIES.iter().cloned().collect(),
         )
-        .map(|responses| {
-            KNOWN_INTROSPECTION_QUERIES
-                .iter()
-                .zip(responses)
-                .filter_map(|(cache_key, response)| match response.into_result() {
-                    Ok(value) => {
-                        let response = Response::builder().data(value).build();
-                        Some((cache_key.into(), response))
-                    }
-                    Err(e) => {
-                        let errors = e
-                            .iter()
-                            .map(std::string::ToString::to_string)
-                            .collect::<Vec<_>>()
-                            .join("\n");
-                        tracing::warn!("introspection returned errors: \n {}", errors);
-                        None
-                    }
+        .map_err(|deno_runtime_error| {
+            tracing::warn!(
+                "router-bridge returned a deno runtime error: \n {}",
+                deno_runtime_error
+            );
+        })
+        .map(|global_introspection_result| {
+            global_introspection_result
+                .map_err(|general_introspection_error| {
+                    tracing::warn!(
+                        "introspection returned an error: \n {}",
+                        general_introspection_error
+                    );
+                })
+                .map(|responses| {
+                    KNOWN_INTROSPECTION_QUERIES
+                        .iter()
+                        .zip(responses)
+                        .filter_map(|(cache_key, response)| match response.into_result() {
+                            Ok(value) => {
+                                let response = Response::builder().data(value).build();
+                                Some((cache_key.into(), response))
+                            }
+                            Err(graphql_errors) => {
+                                let errors = graphql_errors
+                                    .iter()
+                                    .map(std::string::ToString::to_string)
+                                    .collect::<Vec<_>>()
+                                    .join("\n");
+                                tracing::warn!("introspection returned errors: \n {}", errors);
```

Comment:
:yarn: You should probably log each error using a separate log

```suggestion
                                for error in graphql_errors {
                                    tracing::warn!("Introspection returned error: {}", error);
                                }
```

Reply:
fixed in [30c6082](https://github.com/apollographql/router/pull/281/commits/30c608282f2c355d696daf01844f2c88936f9f84)